src/trip-repo.js

```diff
@@ -3,51 +3,95 @@ import Trip from "./trip.js";
   }
 
   compareDates(d1, d2) {
-    const date1 = new Date(d1)
+    const date1 = new Date(d1);
     const date2 = new Date(d2);
-
     return date1 > date2 ? true : false;
   }
 
+  matchDestinationNames(tripArray) {
+    const linkDestination = tripArray.map(trip => {
+      const findDestination = this.destinationData.find(dest => dest.id === trip.destinationID);
+      const newTrip = new Trip(trip, findDestination);
+      return newTrip;
+    });
+    return linkDestination;
+  }
+
+  createOnlyDateDestination(tripArray, userID) {
+    const currentUserTrips = tripArray.filter(trip => trip.userID === userID);
+    const destinationArray = this.matchDestinationNames(currentUserTrips);
+    const newObject = destinationArray.map(trip => ({'date':trip.date, 'destination':trip.destination.destination}));
+    return newObject;
+  }
+
   findUserPastTrips(userID, date) {
-    const goodDates = [];
-    const dateDestinationArray = [];
+    // QUESTION FOR HUGH: I want to return the element because those are the objects I need but when I return it it comes back as False/True. Is this because my matchDestionNames returns only true/false? I also tried to make this chunk of code into its own function but flippting the date and element.date got really funky...
+
+    // const check = this.compareDates(this.allTrips.filter(element => {
+    //   if (this.compareDates(date, element.date)) {
+    //     console.log(element)
+    //     return element;
+    //   } 
+    // }));
+    
+    const pastTrips = [];
     
     this.compareDates(this.allTrips.forEach(element => {
       if (this.compareDates(date, element.date)) {
-        goodDates.push(element)
+        pastTrips.push(element);
       }
     }));
-    const currentUserTrips = goodDates.filter(trip => trip.userID === userID);
-    const destinationArray = this.matchDestinationNames(currentUserTrips);
-    destinationArray.forEach(element => {
-      dateDestinationArray.push({ 'date' : element.date, 'destination' : element.destination.destination })
-    });
-    return dateDestinationArray;
+
+    return this.createOnlyDateDestination(pastTrips, userID);
   }
 
-  matchDestinationNames(tripArray) {
-    const checkArray = [];
-    tripArray.forEach(element => {
-      const newDestination = this.destinationData.filter(dest => dest.id === element.id);
-      const newTrip = new Trip(element, newDestination[0]);
-      checkArray.push(newTrip);
-    });
-    return checkArray;
+  findUserCurrentTrip(userID, date) {
+    const currentUserTrips = this.allTrips.filter(trip => trip.date === date);
+    return this.createOnlyDateDestination(currentUserTrips, userID);
   }
 
-  // findUserCurrentTrip(userID, date) {
-  // }
+  findUserUpcomingTrips(userID, date) {
+    const futureTrips = [];
+    
+    this.compareDates(this.allTrips.forEach(element => {
+      if (this.compareDates(element.date, date)) {
+        futureTrips.push(element);
+      }
+    }));
```

Comment:
You can rewrite this to use `Array.filter` versus `Array.forEach` with a `push`. Also, I see you passing the array to `this.compareDates` here too. Is this intentional? As far as I can tell, `this.compareDates` accepts two dates, not an array.